File: Controle Arduino Desktop/Controle Arduino Desktop/ConnectionApi.vb
Imports System.IO
Imports System.Net
Imports System.Runtime.Serialization.Json
Imports System.Text
Imports Newtonsoft.Json
Public Class ConnectionApi

    Public Shared Function LedApi(ByVal url As String) As Led

        REM Shared baseUrl As String = Uri
        Dim client = New WebClient()
        client.Headers.Add("User-Agent", "Nobody")


        Try
            Dim response = client.DownloadString(New Uri(url))
            Dim led As Led = JsonConvert.DeserializeObject(Of Led)(response)
            Return led
        Catch ex As Exception
            Throw ex
        End Try
    End Function

    Public Shared Function BuzzerApi(ByVal url As String) As Buzzer

        REM Shared baseUrl As String = Uri
        Dim client = New WebClient()
        client.Headers.Add("User-Agent", "Nobody")

        Try
            Dim response = client.DownloadString(New Uri(url))
            Dim buzzer As Buzzer = JsonConvert.DeserializeObject(Of Buzzer)(response)
            Return buzzer
        Catch ex As Exception
            Throw ex
        End Try
    End Function

End Class

File: Controle Arduino Desktop C-SHARP/Controle Arduino Desktop C-SHARP/ConnectionApi.cs
using Controle_Arduino_Desktop_C_SHARP.Models;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Text;
using System.Threading.Tasks;

namespace Controle_Arduino_Desktop_C_SHARP
{
    class ConnectionApi
    {

        public Led LedApi(String url) {


            var client = new WebClient();
            client.Headers.Add("User-Agent", "Nobody");


            try
            {
                var response = client.DownloadString(new Uri(url));
                var led = (Led)JsonConvert.DeserializeObject<Led>(response);
                return led;
    
            }
        catch( Exception ex ) {
                throw ex;
        }
        
        }

        public Buzzer BuzzerApi(String url) {


            var client = new WebClient();
            client.Headers.Add("User-Agent", "Nobody");

            try
            {
                var response = client.DownloadString(new Uri(url));
                var buzzer = (Buzzer)JsonConvert.DeserializeObject<Buzzer>(response);
                return buzzer;
            }
            catch ( Exception ex ) {
                throw ex;
            }

            


        }

    }
}
